rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionInstant.java

```diff
@@ -115,17 +114,11 @@ private String getReturnMessage(String txHash) {
         ProgramResult programResult = this.blockExecutor.getProgramResult(hash);
 
         if (programResult != null && programResult.isRevert()) {
-            Optional<String> revertReason = EthModule.decodeRevertReason(programResult);
-
-            if (revertReason.isPresent()) {
-                throw RskJsonRpcRequestException.transactionRevertedExecutionError(revertReason.get());
-            } else {
-                throw RskJsonRpcRequestException.transactionRevertedExecutionError();
-            }
+            throw transactionRevertedExecutionError(EthModule.decodeProgramRevert(programResult));
         }
 
         if (!transactionInfo.getReceipt().isSuccessful()) {
-            throw transactionRevertedExecutionError();
+            throw RskJsonRpcRequestException.unknownTransactionExecutionError();
```

Comment:
I believe it should still be `transactionRevertedExecutionError()`, even if it was triggered not via the `REVERT` opcode. This change potentially can break existing integrations